src/sentry/tagstore/snuba/backend.py

```diff
@@ -389,105 +417,8 @@ def get_groups_user_counts(self, project_id, group_ids, environment_id):
         aggregations = [['uniq', 'user_id', 'count']]
 
         result = snuba.query(start, end, ['issue'], None, filters, aggregations)
-        return defaultdict(int, result.items())
+        return defaultdict(int, {k: v for k, v in result.items() if v})
 
     def get_group_ids_for_search_filter(
             self, project_id, environment_id, tags, candidates=None, limit=1000):
         raise NotImplementedError
-
-    # Everything from here down is basically no-ops
```

Comment:
I removed these since they should fall through to the `NotImplementedError` in production (otherwise we messed something up pretty bad.) I'd still like to find a clearer way to split the read/write interfaces but handwaving that away into the delegator selector function for now.

Reply:
don't we still want to leave the `get_or_create` ones implemented here? I don't know about al the callers, but it seems like it would be fine if these methods still returned a valid result without creating anything

Reply:
Looking into this, will follow up with a more informed comment.

Reply:
Yeah I'm happy with that answer.

Reply:
Yeah,

* I think we have to fix tests to deal with "actual events" so that different abstractions (tagstore postgres vs tagstore snuba) can do the right thing.
* We shouldn't impl `get_or_create` (or any `create`?) methods in snuba.